kroxylicious-test-tools/src/test/java/io/kroxylicious/test/tester/DefaultKroxyliciousTesterTest.java

```diff
@@ -0,0 +1,146 @@
+            verify(clientFactory).build(eq("demo"), anyString());
+            verify(kroxyliciousClients).admin();
+        }
+    }
+
+    @SuppressWarnings("resource")
+    @Test
+    void shouldCreateAdminForDefaultVirtualCluster(KafkaCluster backingCluster) {
+        // Given
+        try (var tester = buildTester(backingCluster)) {
+
+            // When
+            tester.admin();
+
+            // Then
+            //In theory the bootstrap address is predicable but asserting it is  not part of this test
+            verify(clientFactory).build(eq("demo"), anyString());
+            verify(kroxyliciousClients).admin();
+        }
+    }
+
+    @SuppressWarnings("resource")
+    @Test
+    void shouldCreateProducerForVirtualCluster(KafkaCluster backingCluster) {
+        // Given
+        try (var tester = buildTester(backingCluster)) {
+
+            // When
+            tester.admin("demo");
+
+            // Then
+            //In theory the bootstrap address is predicable but asserting it is  not part of this test
+            verify(clientFactory).build(eq("demo"), anyString());
+            verify(kroxyliciousClients).admin();
+        }
+    }
+
+    @SuppressWarnings("resource")
+    @Test
+    void shouldCreateProducerForDefaultVirtualCluster(KafkaCluster backingCluster) {
+        // Given
+        try (var tester = buildTester(backingCluster)) {
+
+            // When
+            tester.admin();
+
+            // Then
+            //In theory the bootstrap address is predicable but asserting it is  not part of this test
+            verify(clientFactory).build(eq("demo"), anyString());
+            verify(kroxyliciousClients).admin();
+        }
+    }
+
+    @SuppressWarnings("resource")
+    @Test
+    void shouldCreateConsumerForVirtualCluster(KafkaCluster backingCluster) {
+        // Given
+        try (var tester = buildTester(backingCluster)) {
+
+            // When
+            tester.admin("demo");
+
+            // Then
+            //In theory the bootstrap address is predicable but asserting it is  not part of this test
+            verify(clientFactory).build(eq("demo"), anyString());
+            verify(kroxyliciousClients).admin();
+        }
+    }
+
+    @SuppressWarnings("resource")
+    @Test
+    void shouldCreateConsumerForDefaultVirtualCluster(KafkaCluster backingCluster) {
+        // Given
+        try (var tester = buildTester(backingCluster)) {
+
+            // When
+            tester.admin();
+
+            // Then
+            //In theory the bootstrap address is predicable but asserting it is  not part of this test
+            verify(clientFactory).build(eq("demo"), anyString());
+            verify(kroxyliciousClients).admin();
+        }
+    }
+
+    @NotNull
+    private DefaultKroxyliciousTester buildTester(KafkaCluster backingCluster) {
+        return new DefaultKroxyliciousTester(proxy(backingCluster),
```

Comment:
I'm finding the combination of a real backing cluster, a real proxy and a mocked client/clientFactory a bit of odd stack.  A client offers access to a cluster.  So if the client is mock, what's the point of the real kafka cluster, since there will never be a client actually connected to it.

Reply:
Yeah that's fair.

Initially I was planning to actually test the creation of topics. Then when I broke the work out that stopped being a thing to test and I realised all I cared about was creation/calling close.

So would it be better to:
-  mock the cluster as well (and cover what ever it needs to provide the tester)
- assert the returned clients are not null and different instances? I don't think we can assert much else about the client.

Reply:
My gut feel is to aim for a pure unit test i.e. mock the cluster as well.
It might make sense to add integration tests too, but I'd defer that until decision.